src/ophyd_async/testing/_assert.py

```diff
@@ -144,29 +149,27 @@ def __eq__(self, value):
 
 
 class MonitorQueue(AbstractContextManager):
-    def __init__(self, signal: SignalR):
+    def __init__(
+        self, signal: SignalR, timestamp_provider: Callable[[], float] | None = None
+    ):
         self.signal = signal
         self.updates: asyncio.Queue[dict[str, Reading]] = asyncio.Queue()
-        self.signal.subscribe(self.updates.put_nowait)
+        self._timestamp_provider = timestamp_provider or time.time
 
     async def assert_updates(self, expected_value):
         # Get an update, value and reading
-        expected_type = type(expected_value)
-        expected_value = approx_value(expected_value)
         update = await self.updates.get()
-        value = await self.signal.get_value()
-        reading = await self.signal.read()
-        # Check they match what we expected
-        assert value == expected_value
-        assert type(value) is expected_type
-        expected_reading = {
+        # reading = await self.signal.read()
+        await assert_value(self.signal, expected_value)
+        expected_reading: dict[str, Reading] = {
             self.signal.name: {
                 "value": expected_value,
-                "timestamp": pytest.approx(time.time(), rel=0.1),
+                "timestamp": self._timestamp_provider(),  # type: ignore
```

Comment:
Is this just so we can use `time.monotonic`? If so, I think it's fine to change that just in that function without introducing a timestamp provider...